### ISSUE TITLE
Response to Deferred Query with Accept: multipart/mixed is application/json

### ISSUE THREAD

**Issue by alessbell** [apollographql/router#1656]
**Describe the bug**
When making a request containing a query with a `@defer` directive, e.g.

```graphql
query DeferVariation {
  allProducts {
    delivery {
      ...MyFragment @defer
    }
    sku,
    id
  }
}

fragment MyFragment on DeliveryEstimates {
  estimatedDelivery
  fastestDelivery
}
```

and the request header `Accept: multipart/mixed; deferSpec=20220822, application/json`, the client would expect a response type of `multipart/mixed`, while it is receiving a response of type `application/json`.

This is the above query executed against https://github.com/apollographql/supergraph-demo-fed2/pull/212 in Postman (NB: the default `Accept: */*` header is overridden by the custom Accept header at the bottom; when I uncheck the former the response is the same):

<img width="1440" alt="CleanShot 2022-08-30 at 16 38 51@2x" src="https://user-images.githubusercontent.com/5139846/187541400-b1bd09e6-b7d3-4643-8508-33d67f3c3213.png">

**To Reproduce**
There are detailed reproduction steps in the PR description https://github.com/jpvajda/ac-defer-example/pull/5.

**Expected behavior**
If the `@defer` directive is issued on a fragment in the query and the client indicates it can handle `multipart/mixed` responses, the response will be multipart.

**Comment by glasser** [comment]
I think I see the bug

```
                list.any(|mime| mime.as_ref() == Ok(&multipart_mixed))
```

This Eq check compares to see if the types are exactly the same including parameters. You probably want to check to see if `mime.essence()` (which is the type without the parameters) equals multipart_mixed (and then of course look for `deferSpec` too).

**Comment by prasek** [comment]
[adding `deferSpec` to smoke test 6 in PR 212](https://github.com/apollographql/supergraph-demo-fed2/pull/212/commits/ee73ac2a685fb113e0be322e800b8be2cd9c5842#diff-c59203f52e2be3f028336e33aca53cf69c790f1ce6230cec85a29abdf1040817R155) does result in [the same error](https://github.com/apollographql/supergraph-demo-fed2/runs/8103554712?check_suite_focus=true#step:9:146).

**Comment by abernix** [comment]
Thanks very much for opening this issue @alessbell.  Do you mind confirming what version of the Router you were trying?

_Presumably_ it was `1.0.0-alpha.0`.  It's worth noting that the Router still didn't have any awareness of `deferSpec` as of that version; such support is coming in https://github.com/apollographql/router/pull/1652.  So yes, what @glasser wrote is notable, but you're also sending _more_ than we expect right now with the inclusion of `deferSpec`.

Anyhow, we'll take a look at this today and we'll get there soon!  So close.

**Comment by glasser** [comment]
Sure, my suggestion is that in general ignoring unknown parameters is
probably better than rejecting here, though perhaps you disagree.

**Comment by Geal** [comment]
ignoring unknown parameters should be fine yes, I'll update the filter

**Comment by Geal** [comment]
this should be fixed by #1652
